lib/src/index.rs

```diff
@@ -2130,6 +2171,81 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_walk_revs_filter_by_generation() {
+        let mut index = MutableIndex::full(3);
+        // 9 7
+        // | |
+        // 8 6
+        // |/|
+        // 4 5
+        // | |
+        // 2 3
+        // |/
+        // 1
```

Comment:
I'm curious about a case like this:

```
5
|\
| 4
| |
2 3
|/
1
```

Will 1 match when requesting 2 generations before 5 *and* when requesting 3 generations before 5? I think it should (and I think they will as implemented).

Reply:
Yes, they're basically the same as `5--` and `5---` respectively. Adjusted the test to include these cases.